TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementCheckers.cs

```diff
@@ -22,4 +23,53 @@ public static class SavepointStatementChecker
             }
         }
     }
+
+    internal static class ExecuteImmediateStatementChecker
+    {
+        public static void OnCodeElement(ExecuteImmediateStatement executeImmediateStatement, CodeElementsParser.ExecuteImmediateStatementContext context)
+        {
+            var expressionType = executeImmediateStatement.Expression?.ExpressionType;
+            switch (expressionType)
+            {
+                case SqlExpressionType.Variable:
+                    var variable = (SqlVariable) executeImmediateStatement.Expression;
+                    if (variable.Type != VariableType.HostVariable)
```

Comment:
Remove this part as the spec states

> In SQL PL, an SQL variable, SQL parameter, or transition variable can be used

However when a host variable is used, we have to check that there is no indicator.